### PR TITLE
Move from RSpec internal validity page check to html-proofer (LG-10193)

### PR DESCRIPTION
**Pull request by mitchellhenke**
[LG-10193](https://cm-jira.usa.gov/browse/LG-10193)
[LG-12164](https://cm-jira.usa.gov/browse/LG-12164)

Similar to what we did in https://github.com/18F/identity-site/pull/658 for the brochure site, this PR removes the internal custom RSpec matcher and replaces it with html-proofer.

Many of the changes are converting paths prepend the jekyll baseurl and fixing anchors in hrefs. Going forward, all links will have to be valid if this is merged.

### CODE REVIEW

File: _layouts/base.html [hunk 1]
```diff
@@ -95,7 +95,7 @@
       <header class="usa-header usa-header--extended" role="banner">
           <div class="usa-navbar">
             <div class="usa-logo" id="basic-logo">
-              <a href="{{ site.baseurl }}/" title="Home" aria-label="Home">
```
Comment (aduth) [inline _layouts/base.html:98]: Wouldn't we want to keep `site.baseurl` as much as possible to support links in Federalist previews resolving correctly?

File: .circleci/config.yml [hunk 2]
```diff
@@ -33,7 +33,9 @@ jobs:
           command: |
             make clean
             make test
-
+      - run:
+          name: Run htmlproofer (no external links)
+          command: bundle exec scripts/htmlproofer --internal
```
Comment (mitchellhenke) [inline .circleci/config.yml:38]: The `--internal` flag means we will only check internal links. We've found external ones can be a little flaky.

File: _layouts/base.html (hunk 1 above)
Comment (mitchellhenke) [reply _layouts/base.html:98]: yep. we had mixed usage previously and I saw that `identity-site` wasn't using `site.baseurl` (but those appear to just be a few stragglers). Added 6e862e3 to fix it.

**Review by zachmargolis — APPROVED** [review]

**Review by Sgtpluck — APPROVED** [review]
thank you so much!

**Pull request merged**